src/Extensions/Hosting.Services.Web/Middlewares/ActivityEnrichmentMiddleware.cs

```diff
@@ -11,16 +11,11 @@ namespace Microsoft.Omex.Extensions.Hosting.Services.Web.Middlewares
 	/// <summary>
 	/// Enrich request activity with Result, SubType and Metadata
 	/// </summary>
-	internal class ActivityEnrichmentMiddleware
+	internal class ActivityEnrichmentMiddleware : IMiddleware
```

Comment:
Just for my own understanding, could you please explain why this was causing issues before?

Reply:
I'm not sure that it's the best solution but from my experiments and debugging with middleware resolution it looks like without interface middleware constructor able to resolve only limited amount of objects (like HttpContext and the one passed to Use method), but if your class implements an interface then it's properly resolved from DI. I'm not sure why it's done like this and it's possible that I'm misunderstanding something. 
Here the link to resolution if interface implemented https://github.com/dotnet/aspnetcore/blob/425c196cba530b161b120a57af8f1dd513b96f67/src/Http/Http.Abstractions/src/Extensions/UseMiddlewareExtensions.cs#L46
And here the link to resolution in other cases (only parameters used for resolution) https://github.com/dotnet/aspnetcore/blob/425c196cba530b161b120a57af8f1dd513b96f67/src/Http/Http.Abstractions/src/Extensions/UseMiddlewareExtensions.cs#L92